tests/unit/challenges/test_aws_utils.py

```diff
@@ -0,0 +1,511 @@
+        self.ecs_client.create_cluster(clusterName="Challenge_Cluster")
+        aws_utils.create_service_by_challenge_pk(self.ecs_client, self.challenge, client_token)
+        self.ecs_client = boto3.client("ecs")
+        expected_status_code = HTTPStatus.OK
+        response = aws_utils.update_service_by_challenge_pk(self.ecs_client, self.challenge, num_of_tasks)
+        status_code = response["ResponseMetadata"]["HTTPStatusCode"]
+        self.assertEqual(status_code, expected_status_code)
+        self.assertEqual(self.challenge.workers, num_of_tasks)
+
+    def test_update_service_by_challenge_pk_for_non_existent_service(self):
+        num_of_tasks = 2
+        response = aws_utils.update_service_by_challenge_pk(self.ecs_client, self.challenge2, num_of_tasks)
+        self.assertEqual(response["Error"]["Code"], "ServiceNotFoundException")
+
+
+class TestServiceManager(BaseTestClass):
+    def setUp(self):
+        super(TestServiceManager, self).setUp()
+
+    @mock.patch("challenges.aws_utils.client_token_generator")
+    @mock.patch("challenges.aws_utils.create_service_by_challenge_pk")
+    def test_service_manager_for_new_challenge(self, mock_create, mock_token_gen):
+        mock_token_gen.return_value = "abc123"
+        client_token = "abc123"
+        aws_utils.service_manager(self.ecs_client, self.challenge)
+        mock_create.assert_called_with(self.ecs_client, self.challenge, client_token)
+
+    @mock_ecs
+    @mock.patch("challenges.aws_utils.update_service_by_challenge_pk")
+    def test_service_manager_for_active_challenge(self, mock_update):
+        num_of_tasks = 3
+        client_token = "abc123"
+        self.ecs_client.create_cluster(clusterName="Challenge_Cluster")
+        aws_utils.create_service_by_challenge_pk(self.ecs_client, self.challenge2, client_token)
+        aws_utils.service_manager(self.ecs_client, self.challenge2, num_of_tasks=num_of_tasks)
+        mock_update.assert_called_with(self.ecs_client, self.challenge2, num_of_tasks, False)
+
+
+@mock_ecs
+class BaseAdminCallsClass(BaseTestClass):
+    def setUp(self):
+        super(BaseAdminCallsClass, self).setUp()
+
+        self.challenge3 = Challenge.objects.create(
+            title="Test Challenge 3",
+            creator=self.challenge_host_team,
+            queue="test_queue_3",
+            start_date=timezone.now() - timedelta(days=2),
+            end_date=timezone.now() + timedelta(days=1),
+        )
+        self.ecs_client.create_cluster(clusterName="Challenge_Cluster")
+        self.client_token = "abc123"
+
+    @classmethod
+    def queryset(cls, pklist):
+        queryset = Challenge.objects.filter(pk__in=pklist)
+        queryset = sorted(queryset, key=lambda i: pklist.index(i.pk))
+        return queryset
+
+
+class TestStartWorkers(BaseAdminCallsClass):
+    def setUp(self):
+        super(TestStartWorkers, self).setUp()
+
+    def test_start_workers_1(self):  # challenge2 and challenge3 workers are inactive.
```

Comment:
@KhalidRmb please rename the test method, didn't get the case for which test is written.